jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/MetaInfConfiguration.java

```diff
@@ -720,28 +719,32 @@ public FileVisitResult visitFile(Path f, BasicFileAttributes attrs) throws IOExc
      */
     public Collection<URL> getTlds(URI uri) throws IOException
     {
-        HashSet<URL> tlds = new HashSet<URL>();
-
-        URI jarUri = uriJarPrefix(uri, "!/");
-        URL url = jarUri.toURL();
-        JarURLConnection jarConn = (JarURLConnection)url.openConnection();
-        jarConn.setUseCaches(Resource.getDefaultUseCaches());
-        JarFile jarFile = jarConn.getJarFile();
-        Enumeration<JarEntry> entries = jarFile.entries();
-        while (entries.hasMoreElements())
-        {
-            JarEntry e = entries.nextElement();
-            String name = e.getName();
-            if (name.startsWith("META-INF") && name.endsWith(".tld"))
+        HashSet<URL> tlds = new HashSet<>();
+        try (Resource.Mount mount = Resource.mount(uriJarPrefix(uri, "!/"));
+             Stream<Path> stream = Files.walk(mount.root().getPath()))
+        {
+            Iterator<Path> it = stream
+                .filter(MetaInfConfiguration::isTldFile)
```

Comment:
Add another `.filter(Files::isRegularFile)` before this.
Otherwise this will match directories like `/META-INF/foo.tld`, which is odd.

Reply:
That test should be part of `isTldFile()`, shouldn't it?